pkg/volume/csi/csi_mounter.go

```diff
@@ -252,6 +252,14 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		mountOptions,
 	)
 
+	defer func() {
+		if err != nil {
+			klog.Warningf("reverting NodePublishVolume for [%s] due to failure %q", c.volumeID, err.Error())
+			if unpubErr := csi.NodeUnpublishVolume(ctx, c.volumeID, dir); unpubErr != nil {
+				klog.Error(log("NodeUnpublishVolume failed for [%s]: %v", c.volumeID, unpubErr))
+			}
+		}
+	}()
```

Comment:
See L279

may emove the MountDir, then NodeUnpublish?

I do not think it is reasonable